performance/buildSrc/src/main/kotlin/RegressionsReporter.kt

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+import groovy.lang.Closure
+import org.gradle.api.Action
+import org.gradle.api.DefaultTask
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+
+import com.ullink.slack.simpleslackapi.impl.SlackSessionFactory
+
+import org.jetbrains.report.json.*
+
+import java.io.FileInputStream
+import java.io.IOException
+import java.io.File
+import java.util.concurrent.TimeUnit
+import java.net.HttpURLConnection
+import java.net.URL
+import java.util.Base64
+import java.util.Properties
+
+// Run command line from string.
+fun String.runCommand(workingDir: File = File("."),
+                      timeoutAmount: Long = 60,
+                      timeoutUnit: TimeUnit = TimeUnit.SECONDS): String {
+    return try {
+        ProcessBuilder(*this.split("\\s".toRegex()).toTypedArray())
+                .directory(workingDir)
+                .redirectOutput(ProcessBuilder.Redirect.PIPE)
+                .redirectError(ProcessBuilder.Redirect.PIPE)
+                .start().apply {
+                    waitFor(timeoutAmount, timeoutUnit)
+                }.inputStream.bufferedReader().readText()
+    } catch (e: IOException) {
+        error("Couldn't run command $this")
+    }
+}
+
+// List of commits.
+class CommitsList(data: JsonElement): ConvertedFromJson {
+
+    val commits: List<Triple<String, String, String>>
+
+    init {
+        if (data is JsonObject) {
+            val changesElement = getRequiredField(data, "change")
+            if (changesElement is JsonArray) {
+                commits = arrayToList(changesElement.jsonArray, { index ->
+                    if (getObjectOrNull(index) != null)
```

Comment:
The [`?.let`-idiom](https://blog.philipphauer.de/idiomatic-kotlin-best-practices/#consider-let) looks suitable here:
```
getObjectOrNull(index)?.let {
    Triple(elementToString(getRequiredField(it, "version"), "version"),
        elementToString(getRequiredField(it, "username"), "username"),
        elementToString(getRequiredField(it, "webUrl"), "webUrl")
    )
}
```